### PR TITLE
Feature and Features Goa generation

### PR DESCRIPTION
**Pull request by corinnekrych**

### CODE REVIEW

File: controller/feature.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+}
+
+// Show runs the show action.
+func (c *FeatureController) Show(ctx *app.ShowFeatureContext) error {
+	// FeatureController_Show: start_implement
+
+	// Put your logic here
+
+	// FeatureController_Show: end_implement
+	return nil
```
Comment (xcoulon) [inline controller/feature.go]: you could respond with `OK` for now and add a first test to verify that ? so this PR would be include tests from the start

File: design/api.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	})
+
+	a.Trait("conditional", func() {
+		a.Headers(func() {
+			a.Header("If-Modified-Since", d.String)
+			a.Header("If-None-Match", d.String)
+		})
+	})
+
+	//a.JWTSecurity("jwt", func() {
```
Comment (xcoulon) [inline design/api.go]: we'll need `jwt` to handle the user's token in the requests, so this needs to be commented out

File: design/feature.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+}
+
+// genericLinks defines generic relations links
+var genericLinks = a.Type("GenericLinks", func() {
+	a.Attribute("self", d.String)
+	a.Attribute("related", d.String)
+	a.Attribute("meta", a.HashOf(d.String, d.Any))
+})
+
```
Comment (xcoulon) [inline design/feature.go]: all types above could be put in a `jsonapi.go` file

File: design/feature.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+var featureList = JSONList(
+	"Feature", "Holds the list of Features",
+	feature,
+	nil,
+	nil)
+
+var _ = a.Resource("features", func() {
+	a.BasePath("/api/features")
+	a.Action("list", func() {
+		//a.Security("jwt")
```
Comment (xcoulon) [inline design/feature.go]: we'll need the `jwt` security middleware as explained above.

File: main.go [hunk 5]
```diff
@@ -19,4 +27,35 @@ func main() {
...
+	service := goa.New("feature")
+
+	// Mount middleware
+	service.WithLogger(goalogrus.New(log.Logger()))
+	service.Use(middleware.RequestID())
+	service.Use(gzip.Middleware(9))
+	service.Use(jsonapi.ErrorHandler(service, true))
+	service.Use(middleware.Recover())
+
+	//service.Use(witmiddleware.TokenContext(publicKeys, nil, app.NewJWTSecurity()))
```
Comment (xcoulon) [inline main.go]: this will be needed to handle JWT in the requests

**Comment by corinnekrych** [comment]
closing this one infavor of #3

**Pull request closed without merging**